src/createStreamMiddleware.ts

```diff
@@ -102,7 +103,12 @@ export default function createStreamMiddleware(options: Options = {}) {
   function processResponse(res: PendingJsonRpcResponse<unknown>) {
     const context = idMap[res.id as unknown as string];
     if (!context) {
-      throw new Error(`StreamMiddleware - Unknown response id "${res.id}"`);
+      // In case response is received and corresponsing request is not present in idMap
+      // we only show warning and not throw error anymore.
+      // Reason for change is that due to MV3 we have added action replay capability
+      // Thus it can happen that a request ends up being submitted upto 3 retries and there are multiple responses
```

Comment:
Currently we will only retry if the service worker restarts, right? And if that's the case, the previous retry will never return with a response because the process will have died. When would we see multiple responses for the same request?

Reply:
There is a sequence of sync events that come into play in extension repo after which retry message is sent to provider. Though the probability is very very rare but provider may receive multiple retry messages. Thus us better to not throw exception for multiple responses.

Also, I am wondering if exception has any significance in this case ?

Reply:
Good question. I'm not sure if it has significance either. I'm not opposed to switching this to a warning, I just wanted to understand this edge case better.

Reply:
If we're seeing this happen in practice, it might indicate that we're accidentally dispatching the reconnect event multiple times. I can't think of any other way this could happen.

Reply:
Yep I agree, in fact if I revert this change it works well.

Let me revert this change for now and I will wait for more feedback to decide if this change is needed.